src/character.cpp

```diff
@@ -5226,6 +5226,12 @@ hp_part Character::body_window( const std::string &menu_header,
         } else if( limb_is_broken ) {
             desc += colorize( _( "It is broken.  It needs a splint or surgical attention." ), c_red ) + "\n";
             hp_str = "==%==";
+        } else if( has_trait( trait_NOPAIN ) ) {
+            if( current_hp < maximal_hp / 2 ) {
```

Comment:
It should probably be less coarse than this. Having no pain doesn't mean one can't notice large wounds on their body by sight.